lib/web/machineid_test.go

```diff
@@ -722,3 +723,53 @@ func TestListBotInstancesWithSearchTermFilter(t *testing.T) {
 		})
 	}
 }
+
+func TestGetBotInstance(t *testing.T) {
+	ctx := context.Background()
+	env := newWebPack(t, 1)
+	proxy := env.proxies[0]
+	pack := proxy.authPack(t, "admin", []types.Role{services.NewPresetEditorRole()})
+	clusterName := env.server.ClusterName()
+
+	botName := "test-bot"
+	instanceId := uuid.New().String()
+
+	_, err := env.server.Auth().CreateBotInstance(ctx, &machineidv1.BotInstance{
+		Kind:    types.KindBotInstance,
+		Version: types.V1,
+		Spec: &machineidv1.BotInstanceSpec{
+			BotName:    botName,
+			InstanceId: instanceId,
+		},
+		Status: &machineidv1.BotInstanceStatus{},
+	})
+	require.NoError(t, err)
+
+	endpoint := pack.clt.Endpoint(
+		"webapi",
+		"sites",
+		clusterName,
+		"machine-id",
+		"bot",
+		botName,
+		"bot-instance",
+		instanceId,
+	)
+	response, err := pack.clt.Get(ctx, endpoint, url.Values{})
+	require.NoError(t, err)
+	assert.Equal(t, http.StatusOK, response.Code(), "unexpected status code")
+
+	var resp GetBotInstanceResponse
+	require.NoError(t, json.Unmarshal(response.Bytes(), &resp), "invalid response received")
+
+	require.Empty(t, cmp.Diff(resp.BotInstance, machineidv1.BotInstance{
+		Kind:    types.KindBotInstance,
+		Version: types.V1,
+		Spec: &machineidv1.BotInstanceSpec{
+			BotName:    botName,
+			InstanceId: instanceId,
+		},
+		Status: &machineidv1.BotInstanceStatus{},
+	}, protocmp.Transform(), protocmp.IgnoreFields(&machineidv1.BotInstance{}, "metadata")))
+	assert.Equal(t, resp.Yaml, fmt.Sprintf("kind: bot_instance\nmetadata:\n  name: %[1]s\n  revision: %[2]s\nspec:\n  bot_name: test-bot\n  instance_id: %[1]s\nstatus: {}\nversion: v1\n", instanceId, resp.BotInstance.Metadata.Revision))
```

Comment:
```suggestion
	assert.YAMLEq(t, resp.Yaml, fmt.Sprintf("kind: bot_instance\nmetadata:\n  name: %[1]s\n  revision: %[2]s\nspec:\n  bot_name: test-bot\n  instance_id: %[1]s\nstatus: {}\nversion: v1\n", instanceId, resp.BotInstance.Metadata.Revision))
```

This will be slightly more robust.

Reply:
Thanks. Done in [b7ef4a9](https://github.com/gravitational/teleport/pull/55055/commits/b7ef4a91178cffcad1ce18097fbc49c1e2556dcb).